packages/client-python/tests/test_runtime.py

```
@@ -0,0 +1,541 @@
+    def test_initial_state(self):
+        mgr = EngineManager()
+        assert mgr.we_started is False
+        assert mgr.uri is None
+
+    def test_uri_property(self):
+        mgr = EngineManager()
+        mgr._port = 5565
+        assert mgr.uri == 'http://127.0.0.1:5565'
+
+    @pytest.mark.asyncio
+    async def test_ensure_running_reuses_existing(self, tmp_path):
```

Ruff (0.15.7)

[warning] Unused method argument: `tmp_path`

(ARG002)